lib/filewatch/tail_mode/handlers/base.rb

```diff
@@ -43,12 +43,18 @@ def read_to_eof(watched_file)
         begin
           data = watched_file.file_read(@settings.file_chunk_size)
           lines = watched_file.buffer_extract(data)
-          logger.warn("read_to_eof: no delimiter found in current chunk") if lines.empty?
+          if lines.empty?
```

Comment:
what do you think of moving `log_delimiter_not_found` to `watched_file.buffer_extract` ? This way we could remove the duplication of defining `log_delimiter_not_found`. Looking at the information presented by this method, it all comes from the watched_file or the data, so nothing external is necessary.

Reply:
👍 makes sense. Ta.

Reply:
as discussed in another medium, moving the logging operation to  buffer_extract makes the caller information less relevant. For that reason I'm good with not DRYing this piece of code.